ledger/simulation/resources.go

```diff
@@ -345,6 +421,22 @@ func (a *ResourceTracker) addLocal(addr basics.Address, aid basics.AppIndex) boo
 	return true
 }
 
+// Simplify makes the ResourceTracker easier to consume for callers.  It avoids
+// returning boxes "named" by a newly created app.  Those app IDs would not be
+// usable in a later submission. An empty ref can stand in. Once Simplified, the
+// tracker should not be used to for further tracking.
+func (a *ResourceTracker) Simplify() {
```

Comment:
how do callers know when to call this?

Reply:
We call it in the rest API if this consensus update has passed (though see Pavel's comment - perhaps we should do so always.)